codelabs/solutions/kayenta/pipelines/automated-canary.json

```diff
@@ -0,0 +1,431 @@
+                "containers": [
+                  {
+                    "env": [
+                      {
+                        "name": "SUCCESS_RATE",
+                        "valueFrom": {
+                          "configMapKeyRef": {
+                            "key": "SUCCESS_RATE",
+                            "name": "sampleapp-config-canary"
+                          }
+                        }
+                      }
+                    ],
+                    "image": "gcr.io/spinnaker-marketplace/sampleapp:latest",
+                    "name": "sampleapp",
+                    "ports": [
+                      {
+                        "containerPort": 8000
+                      },
+                      {
+                        "containerPort": 8080
+                      }
+                    ]
+                  }
+                ]
+              }
+            }
+          }
+        }
+      ],
+      "moniker": {
+        "app": "sampleapp"
+      },
+      "name": "Deploy Canary",
+      "refId": "3",
+      "relationships": {
+        "loadBalancers": [],
+        "securityGroups": []
+      },
+      "requiredArtifactIds": [
+        "2d1febb3-6b7a-42d4-bcf1-85199c3b1aad"
+      ],
+      "requisiteStageRefIds": [
+        "10",
+        "8"
+      ],
+      "source": "text",
+      "type": "deployManifest"
+    },
+    {
+      "account": "my-kubernetes-account",
+      "cloudProvider": "kubernetes",
+      "manifestArtifactAccount": "embedded-artifact",
+      "manifestArtifactId": "0ac4fa1a-4629-4901-94a5-c5579c775736",
+      "manifests": [
+        {
+          "apiVersion": "apps/v1",
+          "kind": "Deployment",
+          "metadata": {
+            "labels": {
+              "app": "sampleapp-baseline"
+            },
+            "name": "sampleapp-baseline",
+            "namespace": "default"
+          },
+          "spec": {
+            "replicas": 1,
+            "selector": {
+              "matchLabels": {
+                "app": "sampleapp",
+                "version": "baseline"
+              }
+            },
+            "template": {
+              "metadata": {
+                "annotations": {
+                  "prometheus.io/scrape": "true"
+                },
+                "labels": {
+                  "app": "sampleapp",
+                  "version": "baseline"
+                }
+              },
+              "spec": {
+                "containers": [
+                  {
+                    "env": [
+                      {
+                        "name": "SUCCESS_RATE",
+                        "valueFrom": {
+                          "configMapKeyRef": {
+                            "key": "SUCCESS_RATE",
+                            "name": "sampleapp-config-prod"
+                          }
+                        }
+                      }
+                    ],
+                    "image": "gcr.io/spinnaker-marketplace/sampleapp:latest",
```

Comment:
How does someone w/o access to spinnaker marketplace build this image/run this pipeline?

Reply:
`gcr.io/spinnaker-marketplace` is public. Not sure why someone wouldn't have access to this image?

The code for this image is in the app directory, in the same PR.

Reply:
Well they can't push to the repo if they want to make changes or run this themselves -- if that's the intent then this LGTM

Reply:
That's the intent. They always run the same image in the tutorial. It's configurable with an env variable to change the error rate and simulate different behaviours for Kayenta.